src/Mvc/Mvc.Analyzers/src/DiagnosticDescriptors.cs

```diff
@@ -53,7 +53,15 @@ public static class DiagnosticDescriptors
                 isEnabledByDefault: true,
                 helpLinkUri: "https://aka.ms/AA20pbc");
 
-        
         // MVC1005 reserved for startup
+
+        public static readonly DiagnosticDescriptor MVC1006_FunctionsContainingTagHelpersMustBeAsyncAndReturnTask =
+            new DiagnosticDescriptor(
+                "MVC1006",
+                "Functions containing TagHelpers must be async and return Task.",
```

Comment:
super nit: Should `Functions` -> `Methods`? What does the compiler do?

Reply:
L:ooks like C# uses `methods`, will change.